aspnet/mvc/views/forms/sample/final/Controllers/HomeController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using FormsTagHelper.ViewModels;
+using Microsoft.AspNet.Mvc;
+
+namespace FormsTH
+{
+    public class HomeController : Controller
+    {
+        public IActionResult Index()
+        {
+            var model = new CountryViewModel();
+            model.Country = "CA";
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Index(CountryViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                string message = model.Country +  " selected";
+                return RedirectToAction("IndexSuccess", new {id=message});
+            }
+
+            // If we got this far, something failed, redisplay form.
```

Comment:
Suggest including this comment just once in the sample, wherever it would first show up in the doc. Also use a semicolon before "redisplay form".